frontend/src/metabase/visualizations/components/settings/ChartSettingHideButton.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useCallback } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { t } from "ttag";
+import cloneDeep from "lodash.clonedeep";
+import _ from "underscore";
+
+import Button from "metabase/core/components/Button";
+import { updateCardVisualizationSettings } from "metabase/query_builder/actions";
+import type { Column } from "metabase-types/types/Dataset";
```

Comment:
We consider these types as legacy ones, there is `DatasetColumn` in `frontend/src/metabase-types/api/dataset.ts`

Reply:
I looked into it and it appears these two types are slightly different. `Column` has a `visibility_type` that we need to check in order to determine whether the column should be `enabled` in [`getDefaultTableColumns`](https://github.com/metabase/metabase/pull/29137/files#diff-f076ade6e57fe985f13b56fcb2bd68bedd406ac523df96b35018c022e999b222R75).

`DatasetColumn` does not have this type.

Reply:
Yes, that's fine, the newer types miss some fields that we haven't added yet. Feel free to extend them